test/forkMainnet/AMMStrategy.t.sol

```diff
@@ -0,0 +1,404 @@
+        _baseTest(inputToken, outputToken, inputAmount, data);
+    }
+
+    function testTradeCurveV2WithETH() public {
+        address inputToken = Constant.ETH_ADDRESS;
+        address outputToken = WBTC_ADDRESS;
+        uint256 inputAmount = 1 ether;
+        // address[] TRICRYPTO2POOL_COINS = [USDT_ADDRESS, WBTC_ADDRESS, WETH_ADDRESS];
+        int128 inputTokenIndex = 2; // WETH
+        int128 outputTokenIndex = 1; // WBTC
+        IAMMStrategy.Operation[] memory operations = new IAMMStrategy.Operation[](1);
+        // ICurveFiV2
+        bytes memory payload0 = abi.encodeWithSignature(
+            "exchange(uint256,uint256,uint256,uint256,bool)",
+            uint256(uint128(inputTokenIndex)),
+            uint256(uint128(outputTokenIndex)),
+            inputAmount,
+            0,
+            true // use_eth = true
+        );
+        operations[0] = IAMMStrategy.Operation(CURVE_TRICRYPTO2_POOL_ADDRESS, inputAmount, payload0);
+
+        bytes memory data = abi.encode(operations);
+        _baseTest(inputToken, outputToken, inputAmount, data);
+    }
+
+    function testTradeCurveV2WithWETH() public {
+        address inputToken = WETH_ADDRESS;
+        address outputToken = WBTC_ADDRESS;
+        uint256 inputAmount = 10 * 1e18;
+        // address[] TRICRYPTO2POOL_COINS = [USDT_ADDRESS, WBTC_ADDRESS, WETH_ADDRESS];
+        int128 inputTokenIndex = 2; // WETH
+        int128 outputTokenIndex = 1; // WBTC
+        IAMMStrategy.Operation[] memory operations = new IAMMStrategy.Operation[](1);
+        // ICurveFiV2
+        bytes memory payload0 = abi.encodeWithSignature(
+            "exchange(uint256,uint256,uint256,uint256,bool)",
+            uint256(uint128(inputTokenIndex)),
+            uint256(uint128(outputTokenIndex)),
+            inputAmount,
+            0,
+            false // use_eth = false
+        );
+        operations[0] = IAMMStrategy.Operation(CURVE_TRICRYPTO2_POOL_ADDRESS, 0, payload0);
+
+        bytes memory data = abi.encode(operations);
+        _baseTest(inputToken, outputToken, inputAmount, data);
+    }
+
+    function _baseTest(
+        address inputToken,
+        address outputToken,
+        uint256 inputAmount,
+        bytes memory data
+    ) internal {
+        BalanceSnapshot.Snapshot memory inputTokenBalance = BalanceSnapshot.take(entryPoint, inputToken);
+        BalanceSnapshot.Snapshot memory outputTokenBalance = BalanceSnapshot.take(entryPoint, outputToken);
+
+        if (inputToken == Constant.ETH_ADDRESS) {
+            IStrategy(ammStrategy).executeStrategy{ value: inputAmount }(inputToken, outputToken, inputAmount, data);
+        } else {
+            IERC20(inputToken).safeTransfer(address(ammStrategy), inputAmount);
+            IStrategy(ammStrategy).executeStrategy(inputToken, outputToken, inputAmount, data);
+        }
+
+        inputTokenBalance.assertChange(-int256(inputAmount));
+        outputTokenBalance.assertChangeGt(0);
+    }
+
+    function setBalance(
```

Comment:
User `setEOABalanceAndApprove()` and `deal()` instead. See `test/utils/BalanceUtil.sol`

Reply:
6a2cdef